### PR TITLE
Add retry for zip and war deploy

### PR DESCRIPTION
**Pull request by gavinfish**
Related issue #41

### CODE REVIEW

File: src/main/java/com/microsoft/jenkins/appservice/commands/FileDeployCommand.java [hunk 1]
```diff
@@ -63,16 +65,33 @@ public void execute(IFileDeployCommandData context) {
...
+                                return;
+                            } catch (Exception e) {
+                                context.logStatus(String.format("Exception occurred when deploying "
+                                                + "the zip package: %s, retrying immediately (%d/%d)", e.getMessage(),
+                                        retryCount, DEFAULT_MAX_RETRY_TIMES));
+                            }
                         }
                     }
-                    break;
+                    throw new IOException(String.format("The zip deploy failed after %d times of retry.", retryCount));
```
Comment (yungezz) [inline src/main/java/com/microsoft/jenkins/appservice/commands/FileDeployCommand.java]: IOException? is there any other more appriopriate exception type?

File: src/main/java/com/microsoft/jenkins/appservice/commands/FileDeployCommand.java [hunk 2]
```diff
@@ -82,37 +101,53 @@ public void execute(IFileDeployCommandData context) {
...
+                                    break;
+                                } catch (Exception e) {
+                                    context.logStatus(String.format("Exception occurred when deploying "
+                                                    + "the zip package: %s, retrying immediately (%d/%d)",
+                                            e.getMessage(), retryCount, DEFAULT_MAX_RETRY_TIMES));
+                                }
                             }
                         }
+                        if (!deploySuccess) {
+                            throw new IOException(String.format("The war deploy for %s failed after %d times of retry.",
```
Comment (yungezz) [inline src/main/java/com/microsoft/jenkins/appservice/commands/FileDeployCommand.java]: same as above

File: src/test/java/com/microsoft/jenkins/appservice/commands/FileDeployCommandTest.java [hunk 3]
```diff
@@ -24,9 +24,11 @@
 import static org.mockito.ArgumentMatchers.anyString;
 import static org.mockito.ArgumentMatchers.eq;
 import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.never;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.doThrow;
```
Comment (yungezz) [inline src/test/java/com/microsoft/jenkins/appservice/commands/FileDeployCommandTest.java]: seems not sorted. is checkstyle enabled? if  no, pls refer to https://github.com/Microsoft/azure-spring-boot/tree/master/config

File: src/main/java/com/microsoft/jenkins/appservice/commands/FileDeployCommand.java (hunk 1 above)
Comment (gavinfish) [reply src/main/java/com/microsoft/jenkins/appservice/commands/FileDeployCommand.java]: fixed

File: src/main/java/com/microsoft/jenkins/appservice/commands/FileDeployCommand.java (hunk 2 above)
Comment (gavinfish) [reply src/main/java/com/microsoft/jenkins/appservice/commands/FileDeployCommand.java]: fixed

File: src/test/java/com/microsoft/jenkins/appservice/commands/FileDeployCommandTest.java (hunk 3 above)
Comment (gavinfish) [reply src/test/java/com/microsoft/jenkins/appservice/commands/FileDeployCommandTest.java]: fixed temporarily. Will enable checkstyle for test codes in another PR

File: src/main/java/com/microsoft/jenkins/appservice/commands/FileDeployCommand.java [hunk 4]
```diff
@@ -25,6 +26,7 @@
 import java.util.List;
 
 public class FileDeployCommand implements ICommand<FileDeployCommand.IFileDeployCommandData> {
+    public static final int DEFAULT_MAX_RETRY_TIMES = 3;
```
Comment (yungezz) [inline src/main/java/com/microsoft/jenkins/appservice/commands/FileDeployCommand.java]: is this configurable?

Comment (gavinfish) [reply src/main/java/com/microsoft/jenkins/appservice/commands/FileDeployCommand.java]: no, not necessary to expose this to users

**Pull request merged**